### PR TITLE
fix(io): add epoch-settings api and tests

### PR DESCRIPTION
**Pull request by dtfiedler**

### CODE REVIEW

File: src/common/io.ts [hunk 1]
```diff
@@ -124,6 +125,37 @@ export class IOReadable implements AoIORead {
       tags: [{ name: 'Action', value: 'Info' }],
     });
   }
+
+  async getEpochSettings(params?: EpochInput): Promise<AoEpochSettings> {
+    const allTags = [
+      { name: 'Action', value: 'Epoch-Settings' },
+      {
+        name: 'Timestamp',
```
Comment (atticusofsparta) [inline src/common/io.ts:133]: Have you tested this live? "Timestamp" is an overlapping tag I think.

File: src/common/io.ts [hunk 2]
```diff
@@ -124,6 +125,37 @@ export class IOReadable implements AoIORead {
...
+    ];
+
+    const prunedTags: { name: string; value: string }[] = allTags.filter(
+      (tag: {
+        name: string;
+        value: string | undefined;
+      }): tag is { name: string; value: string } => tag.value !== undefined,
+    );
+
+    return this.process.read<AoEpochSettings>({
```
Comment (atticusofsparta) [inline src/common/io.ts:155]: Wondering if process.read should auto-prune tags, think we are repeating this frequently yeah?

File: src/common/io.ts (hunk 1 above)
Comment (dtfiedler) [reply src/common/io.ts:133]: Timestamp is not overlapping on reads

**Pull request merged**